### PR TITLE
reusing existing deployment and service for creating tunnel

### PR DESCRIPTION
**Pull request by chandanpasunoori**
this can useful if user just want use same service ip as it may have configured already in some other deployments/applications to use. 

deleting and recreating service will change ip address for service which may not be desired for few users (for us).

### CODE REVIEW

File: pkg/k8s/exposer.go [hunk 1]
```diff
@@ -52,22 +59,69 @@ func ExposeAsService(namespace, name *string, tunnelPort int, scheme string, raw
...
-		TypeMeta:     v1.TypeMeta{},
-		DryRun:       nil,
-		FieldManager: "",
-	})
-	if err != nil {
-		return err
+
+	var d *appsv1.Deployment
+	var err error
+	deploymentCreated := false
```
Comment (omrikiei) [inline pkg/k8s/exposer.go:65]: when does this value change?

Comment (chandanpasunoori) [reply pkg/k8s/exposer.go:65]: my bad, its supposed to be updated when deployment is created, updated commit.

File: cmd/expose.go [hunk 2]
```diff
@@ -15,6 +15,8 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var ReConnect bool
```
Comment (omrikiei) [inline cmd/expose.go:18]: can we change `-r` and `reconnect` to `-p` and `persist`?

**Review by omrikiei — APPROVED** [review]
approved with comments

File: cmd/expose.go (hunk 2 above)
Comment (chandanpasunoori) [reply cmd/expose.go:18]: make sense, I will update

Comment (chandanpasunoori) [reply cmd/expose.go:18]: @omrikiei -p is conflicting with global flag `-p, --port int` can't use single char alternative for persist

may be `-r, --reuse` ?

Comment (omrikiei) [reply cmd/expose.go:18]: Sure, sorry for misleading you with that

**Comment by chandanpasunoori** [comment]
@omrikiei renamed to reuse and added some error handling if used without reuse flag when resources already exists with same name.

File: cmd/expose.go (hunk 2 above)
Comment (chandanpasunoori) [reply cmd/expose.go]: no problem, updated commit.

**Pull request merged**